platform/linux/mod.rs

```diff
@@ -137,57 +137,65 @@ impl UnixSender {
         }
 
         unsafe {
-            let cmsg_length =
-                (channels.len() + shared_memory_regions.len()) * mem::size_of::<c_int>();
-            let cmsg_buffer = libc::malloc(CMSG_SPACE(cmsg_length as size_t)) as *mut cmsghdr;
-            (*cmsg_buffer).cmsg_len = CMSG_LEN(cmsg_length as size_t);
-            (*cmsg_buffer).cmsg_level = libc::SOL_SOCKET;
-            (*cmsg_buffer).cmsg_type = SCM_RIGHTS;
-
-            let mut fds = Vec::new();
-            for channel in channels.iter() {
-                fds.push(channel.fd());
-            }
-            for shared_memory_region in shared_memory_regions.iter() {
-                fds.push(shared_memory_region.fd);
-            }
-            ptr::copy_nonoverlapping(fds.as_ptr(),
-                                     cmsg_buffer.offset(1) as *mut _ as *mut c_int,
-                                     fds.len());
-            let mut cmsg_padding_ptr =
-                (cmsg_buffer.offset(1) as *mut _ as *mut c_int).offset(fds.len() as isize);
-            let cmsg_end =
-                (cmsg_buffer as *mut _ as *mut u8).offset(CMSG_SPACE(cmsg_length as size_t) as
-                                                          isize);
-            while (cmsg_padding_ptr as *mut u8) < cmsg_end {
-                *cmsg_padding_ptr = *DEV_NULL;
-                cmsg_padding_ptr = cmsg_padding_ptr.offset(1);
-            }
+            let construct_header = |channels: &Vec<UnixChannel>, data_buffer: &Vec<u8>| {
```

Comment:
I wonder whether this would improve readability and/or performance?

(BTW, I intend to make major changes to this code in further iterations -- so probably not worthwhile to spend too much time polishing details... But I can change it of course if it's generally considered better style.)